### PR TITLE
Wine 6.0-rc1 : initial recipe

### PR DESCRIPTION
**Pull request by extrowerk**
ONLY FOR DEVELOPERS, this doesn't builds and doesn't runs your games/programs yet. You have been warned.

A uniform recipe could help us during the porting.

### CODE REVIEW

File: app-emulation/wine/patches/wine-6.0~rc1.patchset [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+diff --git a/configure.ac b/configure.ac
+index 020fd19..c0b393c 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -2201,6 +2201,9 @@ fi
+ dnl Check for -lnsl for Solaris
+ AC_SEARCH_LIBS(gethostbyname, nsl)
+ 
++dnl Check for -lnetwork for Haiku
++AC_SEARCH_LIBS(connect, network)
```
Comment (korli) [inline app-emulation/wine/patches/wine-6.0~rc1.patchset:16]: you can combine the solaris check with the Haiku one
```suggestion
+AC_SEARCH_LIBS(connect, socket network)
```

File: app-emulation/wine/patches/wine-6.0~rc1.patchset [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+index a4b9bc9..c98227d 100644
+--- a/tools/winebuild/build.h
++++ b/tools/winebuild/build.h
+@@ -154,7 +154,8 @@ enum target_platform
+     PLATFORM_APPLE,
+     PLATFORM_FREEBSD,
+     PLATFORM_SOLARIS,
+-    PLATFORM_WINDOWS
++    PLATFORM_WINDOWS,
++    PLATFORM_HAIKU
```
Comment (korli) [inline app-emulation/wine/patches/wine-6.0~rc1.patchset:41]: PLATFORM_HAIKU comes before PLATFORM_SOLARIS

File: app-emulation/wine/patches/wine-6.0~rc1.patchset [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+ #else
+ enum target_platform target_platform = PLATFORM_UNSPECIFIED;
+ #endif
+@@ -135,7 +137,8 @@ static const struct
+     { "solaris", PLATFORM_SOLARIS },
+     { "mingw32", PLATFORM_WINDOWS },
+     { "windows", PLATFORM_WINDOWS },
+-    { "winnt",   PLATFORM_WINDOWS }
++    { "winnt",   PLATFORM_WINDOWS },
++    { "haiku",   PLATFORM_HAIKU }
```
Comment (korli) [inline app-emulation/wine/patches/wine-6.0~rc1.patchset:64]: Comes before PLATFORM_SOLARIS.

File: app-emulation/wine/patches/wine-6.0~rc1.patchset [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+index 6decb02..b8e8f48 100644
+--- a/tools/winegcc/winegcc.c
++++ b/tools/winegcc/winegcc.c
+@@ -191,7 +191,8 @@ static const struct
+     { "mingw32",     PLATFORM_MINGW },
+     { "windows-gnu", PLATFORM_MINGW },
+     { "windows",     PLATFORM_WINDOWS },
+-    { "winnt",       PLATFORM_MINGW }
++    { "winnt",       PLATFORM_MINGW },
++    { "winnt",       PLATFORM_HAIKU }
```
Comment (korli) [inline app-emulation/wine/patches/wine-6.0~rc1.patchset:92]: let's fix this please :)
```suggestion
+    { "haiku",       PLATFORM_HAIKU }
```

File: app-emulation/wine/patches/wine-6.0~rc1.patchset (hunk 1 above)
Comment (extrowerk) [reply app-emulation/wine/patches/wine-6.0~rc1.patchset:16]: Thanks, thats true.

File: app-emulation/wine/patches/wine-6.0~rc1.patchset [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+diff --git a/dlls/ntdll/unix/server.c b/dlls/ntdll/unix/server.c
+index 7236f0a..78fb62c 100644
+--- a/dlls/ntdll/unix/server.c
++++ b/dlls/ntdll/unix/server.c
+@@ -1424,7 +1424,9 @@ void server_init_process(void)
+     /* setup the signal mask */
+     sigemptyset( &server_block_set );
+     sigaddset( &server_block_set, SIGALRM );
++#if !defined(__HAIKU__)
+     sigaddset( &server_block_set, SIGIO );
```
Comment (korli) [inline app-emulation/wine/patches/wine-6.0~rc1.patchset:181]: SIGIO is often a synonym for SIGPOLL

**Comment by extrowerk** [comment]
Status: 
- most files builds
- ntdll/unix/signal*.c needs patching
- somehow we needs to pass -lnetwork to winegcc, as linking fails

**Comment by waddlesplash** [comment]
The signal handling code may need work on the Haiku side, as it needs some registers that we do not appear to pass on x64 (we pass some of them via 32-bit extended vregs, but not x64), but the code might be reworkable to not require these (it appears to have once required even more.)

The other hurdle I didn't know what to do with is that WINE wants to set `gsbase`, which we do not appear to have a way to do. macOS does not officially support it either, so perhaps we can reuse a lot of the code in WINE for that?

**Pull request closed without merging**